src/pages/index.js

```diff
@@ -64,55 +80,182 @@ const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
+/* -------------------------------------------------------------------------- */
+
+// Create the delete popup instance
+const deletePopup = new Popup({ popupSelector: selectors.deletePopup });
+
+deletePopup.setEventListeners();
+
 /* -------------------------------------------------------------------------- */
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
 
 // This function creates a new card
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
-
-    }}, selectors.cardTemplate);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+
+            // Open confirmation popup on click
+            deletePopup.open();
+
+            // Handle card deletion
+            api.deleteCard(data._id)
+            .then(() => {
+
+                // Render loading status
+                renderLoading(true, deleteConfirmButton, 'Deleting...');
+
+                // Remove the card from the page
+                cardElement.deleteCard();
+            })
+            .then(() => {
+                
+                // Close the confirmation popup
+                deletePopup.close();
+            })
+            .catch(err => {
+
+                // If the server returns an error, reject the promise
+                console.error(`Error: ${err.status}`);
+            })
+            .finally(() => {
+
+                // Restore pre-loading status
+                renderLoading(false, deleteConfirmButton, 'Deleting...');
+            })
+        },
+        handleLikeClick: () => {
+
+            // If the user has liked the card, remove the like; vice versa
+            if (cardElement.isLiked()) {
+
+                // Remove like from the server if user has already liked the card
+                api.removeLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.setLikes(card.likes);
+
+                    // Deactivate like icon
+                    cardElement.removeLikeIcon();
+                })
+                .catch(err => {
+
+                    // If the server returns an error, reject the promise
+                    console.error(`Error: ${err.status}`);
+                })
+            } else {
+
+                // Add like to the server if user has not liked the card
+                api.addLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.setLikes(card.likes);
+
+                    // Activate like icon
+                    cardElement.addLikeIcon();
```

Comment:
There is no need to call this method, since you do everything in `setLikes` method

Reply:
It's removed now! Thank you.